File: shared/types/asset.ts
// Asset 类型定义 — 本地资源管理
// 对应 GAME_DESIGN.md §4.1

export type AIModality = 'text' | 'image' | 'video' | 'music' | 'voice';

export type AssetType = 'image' | 'video' | 'audio' | 'voice';

export type AssetSource = 'ai_generated' | 'builtin' | 'local_file';

export interface LocalAsset {
  id: string;
  type: AssetType;
  localPath: string;
  source: AssetSource;
  cacheKey: string;
  createdAt: number;
}

File: src-tauri/src/types/asset.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum AIModality {
    Text,
    Image,
    Video,
    Music,
    Voice,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum AssetType {
    Image,
    Video,
    Audio,
    Voice,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum AssetSource {
    AiGenerated,
    Builtin,
    LocalFile,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LocalAsset {
    pub id: String,
    #[serde(rename = "type")]
    pub asset_type: AssetType,
    pub local_path: String,
    pub source: AssetSource,
    pub cache_key: String,
    pub created_at: u64,
}
